workflows/dgd_fusion_annotator_wf.cwl

```diff
@@ -0,0 +1,70 @@
+cwlVersion: v1.2
+class: Workflow
+id: dgd-fusion-annotator-wf
+label: DGD Fusion Annotator Workflow
+doc: "Formats and annotates DGD fusion files"
+
+
+inputs:
+  # multi
+  output_basename: { type: string, doc: "Basename for the output TSV file" }
+  # update gene symbols
+  hgnc_tsv: { type: 'File?', doc: "Gene name database TSV file from HGNC. i.e. hgnc_complete_set.txt. If not provided, updated will be skipped" }
+  fusions_tsv: { type: File, doc: "Custom Fusions TSV file, i.e. fusion-dgd.tsv.gz" }
+  old_symbol: { type: 'string?', doc: "Column name for the old gene symbol(s) in the HGNC TSV. Set to override script defaults" }
+  new_symbol: { type: 'string?', doc: "Column name for the new gene symbol(s) in the HGNC TSV. Set to override script defaults" }
+  update_columns: {type: 'string?', doc: "Space-separated column names from the Fusions TSV where to update gene names (e.g. -u foo bar blah). Set to override script defaults" }
+  # fusion standardization
+  caller: { type: [{type: enum, name: caller, symbols: ["STARFUSION", "ARRIBA", "DGD", "CUSTOM"]}], doc: "Caller used to produce input",
+    default: "DGD" }
+  # annotate fusion
+  genome_tar: {type: 'File', doc: "STAR-Fusion CTAT Genome lib", "sbg:suggestedValue": {
+      class: File, path: 62853e7ad63f7c6d8d7ae5a8, name: GRCh38_v39_CTAT_lib_Mar242022.CUSTOM.tar.gz}}
+  genome_untar_path: {type: ['null', string], doc: "This is what the path will be when genome_tar is unpackaged", default: "GRCh38_v39_CTAT_lib_Mar242022.CUSTOM"}
+  col_num: {type: ['null', int], doc: "column number in file of fusion name, use 24 for arriba v1.1, 30 for v2, 1 for DGD", default: 1}
+
+outputs:
+  annotated_fusion_tsv: { type: File, outputSource: fusion_annotator/annotated_tsv}
+
+steps:
+  update_gene_symbols:
+    run: ../tools/update_gene_symbols.cwl
+    when: $(inputs.hgnc_tsv != null)
+    in:
+      hgnc_tsv: hgnc_tsv
+      fusions_tsv: fusions_tsv
+      output_filename:
+        source: output_basename
+        valueFrom: |
+          $(self + "_gene_sym_updated.tsv")
+      old_symbol: old_symbol
+      new_symbol: new_symbol
+      update_columns: update_columns
+    out: [updated_fusion_tsv]
+
+  fusion_standardization:
+    run: ../tools/fusion_standardization.cwl
+    in:
+      fusions_tsv:
+        source: [update_gene_symbols/updated_fusion_tsv, fusions_tsv]
+        pickValue: first_non_null
+      caller: caller
+      output_basename: output_basename
+    out: [standardized_fusions]
+
+  fusion_annotator:
+    run: ../tools/fusion_annotator.cwl
+    in:
+      input_fusion_file: fusion_standardization/standardized_fusions
+      genome_tar: genome_tar
+      genome_untar_path: genome_untar_path
+      col_num: col_num
+      output_basename: output_basename
+    out: [annotated_tsv]
+
+$namespaces:
+  sbg: https://sevenbridges.com
+
+hints:
+  - class: 'sbg:AWSInstanceType'
+    value: c5.2xlarge;ebs-gp2;400
```

Comment:
Idk if it's wise to include some sort of reasoning for this choice. Kind of seems out of place.

Reply:
Sure, so, not a big deal at the moment, but it does shave cost for time storage. Time since it's a pretty small wf, can all run on same instance + faster cores. Cost with the time savings and the smaller disk footprint.

Reply:
Yeah I think we talked about the reasoning for this elsewhere but if someone sees it and is like "this seems like a weirdly specific instance choice" it would be nice if we had a doc explaining why.

Reply:
Oddly enough - I was able to add a `doc:` to that field. I think that should do it?

Reply:
Wild. Yeah, i guess that works.